llvm/lib/ProfileData/InstrProf.cpp

```diff
@@ -453,11 +471,94 @@ Error InstrProfSymtab::create(Module &M, bool InLTO) {
     if (Error E = addFuncWithName(F, getPGOFuncName(F, InLTO)))
       return E;
   }
+
+  SmallVector<MDNode *, 2> Types;
+  for (GlobalVariable &G : M.globals()) {
+    if (!G.hasName())
+      continue;
+    Types.clear();
+    G.getMetadata(LLVMContext::MD_type, Types);
+    if (!Types.empty()) {
+      MD5VTableMap.emplace_back(G.getGUID(), &G);
+    }
+  }
   Sorted = false;
   finalizeSymtab();
   return Error::success();
 }
 
+/// \c NameStrings is a string composed of one of more possibly encoded
+/// sub-strings. The substrings are separated by 0 or more zero bytes. This
+/// method decodes the string and calls `NameCallback` for each substring.
+static Error
+readAndDecodeStrings(StringRef NameStrings,
```

Comment:
Good point. Updated InstrProfTest.cpp to have test coverage for the added value type.  There are quite some common code for indirect_call and vtable test coverage, so factor them out. 

I could split the unit-test refactor as a pre-commit of this PR if that's better, but thought to put them here first for the whole picture how they are used.

Test case construction catches an edge case (see the comment in `InstrProfSymtab::getVTableHashFromAddress`), which should be rare at least under Itanium ABI.

Reply:
Since InstrProfTest.cpp changes a lot, created a [nfc patch](https://github.com/llvm/llvm-project/pull/72611/files) to prepare for test coverage of a new type of value profiles.